tests/AppBundle/Service/CasrecVerificationServiceTest.php

```diff
@@ -0,0 +1,201 @@
+            ->shouldReceive('findBy')->with([ 'caseNumber'=>'11111111', 'clientLastname' => 'csurn', 'deputySurname' => 'wrong'])->andReturn([])
+
+            ->shouldReceive('findBy')->with([ 'caseNumber'=>'22222222', 'clientLastname' => 'csurn', 'deputySurname' => 'dsurn'])->andReturn([$crLayNoPC])
+            ->shouldReceive('findBy')->with([ 'caseNumber'=>'33333333', 'clientLastname' => 'csurn', 'deputySurname' => 'mldunique'])->andReturn([$casrecMLD1A])
+            ->shouldReceive('findBy')->with([ 'caseNumber'=>'33333333', 'clientLastname' => 'csurn', 'deputySurname' => 'sibling'])->andReturn([$casrecMLD1B, $casrecMLD1C])
+            ->shouldReceive('findBy')->with([ 'caseNumber'=>'44444444', 'clientLastname' => 'csurn', 'deputySurname' => 'sibling'])->andReturn([$casrecMLD2A, $casrecMLD2B])
+            ->getMock();
+
+        $em = m::mock('\Doctrine\Common\Persistence\ObjectManager')
+            ->shouldIgnoreMissing(true)
+            ->shouldReceive('getRepository')->with('AppBundle\Entity\CasRec')->andReturn($mockCasrecRepo)
+            ->getMock();
+
+        $this->casrecVerificationService = new CasrecVerificationService($em);
+    }
+
+    public function tearDown()
+    {
+        m::close();
+    }
+
+    /**
+     * @test
+     */
+    public function isMultiDeputyCase()
+    {
+        $this->assertFalse($this->casrecVerificationService->isMultiDeputyCase('11111111'));
+        $this->assertTrue($this->casrecVerificationService->isMultiDeputyCase('33333333'));
+        $this->assertTrue($this->casrecVerificationService->isMultiDeputyCase('44444444'));
+    }
+
+    /**
+     * @test
+     */
+    public function validateNonMLDWithPostcode()
+    {
+        $this->assertTrue($this->casrecVerificationService->validate('11111111', 'CSurn', 'DSurn', 'DPC123'));
+
+        // test each fail individually
+        $failMessage = 'User registration: no matching record in casrec';
+        try {
+            $this->casrecVerificationService->validate('WRONG678', 'CSurn', 'DSurn', 'DPC123');
+        } catch(\RuntimeException $e) {
+            $this->assertEquals($failMessage, $e->getMessage());
+        }
+
+        try {
+            $this->assertTrue($this->casrecVerificationService->validate('11111111', 'WRONG', 'DSurn', 'DPC123'));
+        } catch(\RuntimeException $e) {
+            $this->assertEquals($failMessage, $e->getMessage());
+        }
+
+        try {
+            $this->assertTrue($this->casrecVerificationService->validate('11111111', 'CSurn', 'WRONG', 'DPC123'));
+        } catch(\RuntimeException $e) {
+            $this->assertEquals($failMessage, $e->getMessage());
+        }
+
+        try {
+            $this->assertTrue($this->casrecVerificationService->validate('11111111', 'CSurn', 'DSurn', 'WRONG'));
+        } catch(\RuntimeException $e) {
+            $this->assertEquals($failMessage, $e->getMessage());
+        }
```

Comment:
Better to have these as separate tests. One for sucess, one for failures and one testing for exceptions... Pretty good for a first attempt though... instead of using try catch you can use $this->setExpectedException()